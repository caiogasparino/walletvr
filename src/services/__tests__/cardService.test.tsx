// CardService.test.ts
import {CardService} from '@Services';
import axios from 'axios';

jest.mock('axios');

describe('CardService', () => {
  const mockedAxios = axios as jest.Mocked<typeof axios>;

  beforeEach(() => {
    mockedAxios.request.mockReset();
    expect(mockedAxios.request.mockReset()).toMatchSnapshot();
  });

  it('should fetch cards successfully', async () => {
    const cards = [{id: '1', name: 'Card 1'}];

    mockedAxios.request.mockResolvedValueOnce({data: cards, status: 200});

    const cardService = new CardService();

    const result = await cardService.getCards();

    expect(result).toEqual(cards);

    expect(mockedAxios.request).toHaveBeenCalledWith({
      method: 'GET',
      url: 'http://localhost:3000/cards/',
      headers: {'Content-Type': 'application/json'},
    });
  });
});
