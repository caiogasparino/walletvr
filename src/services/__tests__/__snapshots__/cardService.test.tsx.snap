// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardService should fetch cards successfully 1`] = `[MockFunction]`;
